src/com/facebook/buck/android/AndroidLibraryDescription.java

```diff
@@ -47,17 +52,20 @@
 import java.nio.file.Path;
 
 public class AndroidLibraryDescription
-    implements Description<AndroidLibraryDescription.Arg>, Flavored {
+    implements Description<AndroidLibraryDescription.Arg>, Flavored,
+       ImplicitDepsInferringDescription<AndroidLibraryDescription.Arg> {
 
   public static final BuildRuleType TYPE = BuildRuleType.of("android_library");
 
   private static final Flavor DUMMY_R_DOT_JAVA_FLAVOR =
       AndroidLibraryGraphEnhancer.DUMMY_R_DOT_JAVA_FLAVOR;
 
   private final JavacOptions defaultOptions;
+  private final ScalaBuckConfig scalaBuckConfig;
 
-  public AndroidLibraryDescription(JavacOptions defaultOptions) {
+  public AndroidLibraryDescription(JavacOptions defaultOptions, ScalaBuckConfig scalaConfig) {
```

Comment:
What if we had a `JvmLanguagesConfig` interface or base class that the`Scala/Kotlin/BuckConfig` classes implement.  Then we use `JvmLanguagesConfig` here. 

2) Pass in an android compiler instead of a buck config. Your `getCompiler()` implementation doesn't look like it uses anything inside this class and can probably be pulled out into some selector class that choose